docs/ref/models/fields.txt

```diff
@@ -86,14 +86,26 @@ If a field has ``blank=False``, the field will be required.
 
 .. attribute:: Field.choices
 
-A :term:`sequence` consisting itself of iterables of exactly two items (e.g.
-``[(A, B), (A, B) ...]``) to use as choices for this field. If choices are
-given, they're enforced by :ref:`model validation <validating-objects>` and the
-default form widget will be a select box with these choices instead of the
-standard text field.
-
-The first element in each tuple is the actual value to be set on the model,
-and the second element is the human-readable name. For example::
+A mapping or iterable in the format described below to use as choices for this
```

Comment:
I've been reading these edited docs from top to bottom a few times, I do believe they are now polished and ready to go.

One thing that I have just noticed, is that one may define `choices` to be a dict, but then when querying the choices for a Field, because of the normalization call, the result would no longer be the given dict (or Enum for that matter). While this is OK in terms of functionality, I think this section should mention this fact. Perhaps a little down below (see my next comment)?

Reply:
Sounds good. I've amended your suggested text slightly, however.